packages/engine/src/avatar/AvatarInputSystem.ts

```diff
@@ -275,7 +273,7 @@ const execute = () => {
         if (primaryClickCount > 0) {
           primaryClickTimer += getState(EngineState).deltaSeconds
           if (primaryClickTimer <= primaryClickTimeout) {
-            if (buttons.PrimaryClick?.up) {
+            if (buttons.PrimaryClick?.up && isAvatarClicked()) {
```

Comment:
let's store the result of `buttons.PrimaryClick?.up && isAvatarClicked()` so we don't have to recompute it a few lines later

Reply:
please wait, i am testing it

Reply:
its done, and i optimized it quite a bit as well